### PR TITLE
Add azurite recipe

### PR DESCRIPTION
**Pull request by quinnj**

### CODE REVIEW

File: A/azurite/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+# The products that we will ensure are always built.
+products = [
+    FileProduct("azurite/dist/src/azurite.js", :azurite),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    HostBuildDependency("NodeJS_16_jll"),
+    "NodeJS_16_jll",
```
Comment (giordano) [inline A/azurite/build_tarballs.jl:45]:
```suggestion
    Dependency("NodeJS_16_jll"),
```

Comment (quinnj) [reply A/azurite/build_tarballs.jl]: added manually

**Comment by quinnj** [comment]
Hmmm, a bit stuck here; running `julia build_tarballs.jl --debug --verbose x86_64-linux-musl` locally completes successfully, but the azure pipelines CI here are failing across the board. I'll keep poking around and try some things out, but just wondered if @staticfloat might have any ideas on what might be going on here.

**Comment by quinnj** [comment]
updated to a new strategy

**Comment by giordano** [comment]
As far as I can tell, the file `node_modules/azurite/dist/src/azurite.js` is identical across different platforms.  If it is expected that the result is entirely platform-independent, this should use `AnyPlatform`: https://docs.binarybuilder.org/stable/building/#Platform-independent-packages

**Comment by giordano** [comment]
Correct.

**Comment by fredrikekre** [comment]
Bit of a drive by comment, but what is the point of using a jll package for something like this which seems to be just a file in the end?

**Comment by quinnj** [comment]
The `azurite.js` file is just the highest level "entry point" for a whole host of javascript files needed, just like a higher-level Julia package has a bunch of necessary dependencies. So it's convenient to bundle all the necessary files up and provide the `azurite.js` to be accessible and used w/ the NodeJS_jll in order to run this consistently across platforms.

**Comment by quinnj** [comment]
Merge?

**Comment by giordano** [comment]
I wanted to follow up from a question I asked in Slack: if you don't need anything really platform-specific and don't need to use this as dependency of anything else, probably using [`ArtifactUtils.jl`](https://github.com/simeonschaub/ArtifactUtils.jl) is a better option for this, instead of turning Pkg into yet another javascript package manager?

**Comment by quinnj** [comment]
It's not entirely clear how that would look; I want to be able to reuse an azurite_jll package from multiple dependents. So I create that myself and use ArtifactUtils.jl to create the Artifacts.toml and then depend on NodeJS_jll? It just feels like I pretty much need to do what is being done here and this takes care of a lot of the admin for me?

**Comment by quinnj** [comment]
So to be more clear: yes, I'd like to use this as a dependency in several other packages for testing purposes.

File: A/azurite/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+# Set sources and other environment variables.
+name = "azurite"
+version = v"3.18.0"
+sources = GitSource[]
+
+script = "version=$(version)\n" * raw"""
+apk add --update nodejs npm
+cd ${prefix}
+npm install azurite@${version}
```
[the remark this answers is not in the document]
Comment (quinnj) [reply A/azurite/build_tarballs.jl:11]: NodeJS is a _dependency_, but not a `HostBuildDependency`, meaning we need it to _run_ the `azurite.js` `FileProduct`, but not to build it. I tried using it also as a `HostBuildDependency`, but there was some weirdness and this ended up being simpler.

[the remark this answers is not in the document]
Comment (quinnj) [reply A/azurite/build_tarballs.jl:11]: Um......yeah, I think that's technically correct. I was thinking of making an Azurite.jl package that depended on both and provided a slightly higher-level API for running things to make it more convenient.

**Pull request merged**